File: code_examples/3-Structured_Problem_Solving/13-While_Loop/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a while loop
 */

package main

import (
	"fmt"
)

func main() {
	// this function uses a while loop

	var counter int = 0
	var positiveInteger int

	// input
	fmt.Print("Enter how many times to repeat: ")
	fmt.Scan(&positiveInteger)
	fmt.Println()

	// process & output
	for counter < positiveInteger {
		fmt.Printf("%d time(s) through the loop.\n", counter)
		counter++ // a short form for: counter = counter + 1
	}

	fmt.Println("\nDone.")
}

File: code_examples/3-Structured_Problem_Solving/13-While_Loop/CPP/main.cpp
// Copyright (c) 2020 St. Mother Teresa HS All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses a while loop

#include <iostream>

int main() {
    // this function uses a while loop
    int counter = 0;
    int positiveInteger;

    // input
    std::cout << "Enter how many times to repeat: ";
    std::cin >> positiveInteger;
    std::cout << std::endl;

    // process & output
    while (counter < positiveInteger) {
        std::cout << counter << " time(s) through the loop." << std::endl;
        counter = counter + 1;
    }

    std::cout << "\nDone." << std::endl;
    return 0;
}
